File: src/1375-Bulb-Switcher-III/1375.go
func numTimesAllBlue(light []int) int {
    res, maxv := 0, 0
    for i := 0; i < len(light); i++ {
        maxv = max(maxv, light[i])
        if maxv == i + 1 {
            res++
        }
    }
    return res
    
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: src/1375-Bulb-Switcher-III/1375.java
class Solution {
    public int numTimesAllBlue(int[] light) {
        int res = 0, maxv = 0;
        for (int i = 0; i < light.length; i++) {
            maxv = Math.max(maxv, light[i]);
            if (maxv == i + 1) res++;
        }
        return res;
    }
}